### ISSUE TITLE
Feature: Add the ability to search Projects and Users within the App.

### ISSUE THREAD

**Issue by manjotsidhu** [CircuitVerse/mobile-app#41]
**Describe the solution you'd like**
A simple search from where we can find public Projects and Users.

**Are you working on this? No**

**Comment by ZadeAbhishek** [comment]
can i work on this issue.

**Comment by manjotsidhu** [comment]
Sure, go ahead 👍 

> can i work on this issue.

**Comment by manjotsidhu** [comment]
I think Search API is not yet exposed. @Nitish145 knows better.

**Comment by Nitish145** [comment]
Search needs to be implemented, check api.circutverse.org for complete details.

**Comment by ZadeAbhishek** [comment]
okay sure

**Comment by kishanhitk** [comment]
I checked [api.circuitverse.org](api.circuitverse.org) but I can't find any direct search API endpoint for projects or users.
@Nitish145, is getting all users using `/api/v1/users` and doing a search on that list locally a good design?

**Comment by tachyons** [comment]
No, it requires search to be implemented in the api, filtering in client side won't work

**Comment by JatsuAkaYashvant** [comment]
@tachyons, I have confirmed that we have the search functionality for the project. But we still lack the search functionality for the users. A separate issue has been raised #599, to keep things clear.
